### ISSUE TITLE
Readme confusion

### ISSUE THREAD

**Issue by hughfdjackson** [dominictarr/curry#4]
More than an FYI than an issue; after reading the README (well, skimming at least), I don't have a clear the library is for: what do these calls result in?

This certainly doesn't resemble the API i was expecting, e.g.:

```
var add = curry(function(a, b){ return a + b })

add(1)(2) // 3

// plus possibly:
add(1, 2) // 3
```

Perhaps some clearer practical examples would help?

**Comment by dominictarr** [comment]
this module is for partially applying functions, that is, taking a function that takes 3 args (say)
and transforming it into a function that takes 2 args, and returns a function that takes 1 arg.

http://en.wikipedia.org/wiki/Currying

I wouldn't recommend using this module, this trys to be a little clever, and it's simpler to just inline
functions that do stuff like this, rather than create too much abstraction by using a module.

**Comment by hughfdjackson** [comment]
^^ After a re-read, I gleaned that much.  Cheers!

Do you think that the wiki on currying is better satisfied by my suggested API than the current one of the module?  I'm thinking WRT to this section: 

> In mathematics and computer science, currying is the technique of transforming a function that takes multiple 
> arguments (or a tuple of arguments) in such a way that it can be called as a chain of functions, each with a single 
> argument (partial application)

That's most of what caused my confusion; being used to currying from haskell primarily. 

In any case - if you're not currently maintaining this and/or aren't feeling pedantic, this may be moot as far as you're concerned :)

**Comment by dominictarr** [comment]
Yeah, it's probably better - this was one of my first node modules.

If you want to take this over, just say, and I'll add you in github and npm.
bump the major version, so that old versions don't break.

**Comment by hughfdjackson** [comment]
I'd absolutely love to do that :D

**Comment by hayesmaker** [comment]
slurp slurp

**Comment by dominictarr** [comment]
Okay, you are now a collaborator, and an owner on npm. This means you can push to the github repo and publish new versions on npm.

You are free to do what you wish with this module, but please don't break the versions that are already published `0.0.0 - 0.0.3`

So, just publish your first new version as 1.0.0, and we should be all right. there are a few things that depend on this (old things) but I've also used it as an example module for npm related stuff I'm working on.

enjoy!

**Comment by hughfdjackson** [comment]
@hayesmaker you utter troll.

@dominictarr :) great! thanks a bunch.
